### ISSUE TITLE
Limit the allowed characters when registering a number

### ISSUE THREAD

**Issue by ammendonca** [RestComm/Restcomm-Connect#2371]
When registering a number we should perform validation at the API level, not all characters should be allowed (we should only allow what we want and reject anything else).

For instance is possible to register `$` as a number and after that it's not possible to register additional fully numeric numbers and in the server logs, we can see `15:41:48,807 INFO  [org.restcomm.connect.dao.mybatis.MybatisIncomingPhoneNumbersDao] (http-/127.0.0.1:8080-7) Pattern "$" matched the phone number "432423"`, so it seems to be messing with the regex for existing numbers.

**Comment by meghnatolani** [comment]
Hi, can I work on this bug?

**Comment by ammendonca** [comment]
@meghnatolani yes, please. Let me know if you need more clarification on the issue.
